desk/app/contacts.hoon

```diff
@@ -219,63 +294,66 @@
       ++  si-abet
         ^+  cor
         ?-  sas
-          %live  =.  rol  (~(put by rol) who for sag)
+          %live  =.  peers  (~(put by peers) who [for sag])
+                 ?.  new  cor
                  ::  NB: this assumes con.for is only set in +si-hear
                  ::
-                 ?.(new cor (p-news:pub who ~))
-        ::
+                 =.  cor  (p-news-0:pub who ~)
+                 (p-news:pub [%peer who ~])
+          ::
           %dead  ?:  new  cor
-                 =.  rol  (~(del by rol) who)
+                 =.  peers  (~(del by peers) who)
                  ::
                  ::  this is not quite right, reflecting *total* deletion
                  ::  as *contact* deletion. but it's close, and keeps /news simpler
                  ::
-                 (p-news:pub who ~)
+                 =.  cor  (p-news-0:pub who ~)
+                 (p-news:pub [%peer who ~])
         ==
       ::
       ++  si-take
-        |=  =sign:agent:gall
+        |=  [=wire =sign:agent:gall]
         ^+  si-cor
         ?-  -.sign
           %poke-ack   ~|(strange-poke-ack+wire !!)
         ::
           %watch-ack  ~|  strange-watch-ack+wire
-                      ?>  ?=(%want sag)
-                      ?~  p.sign  si-cor(sag [%chi ~])
-                      %-  (slog 'contact-fail' u.p.sign)
-                      pe-peer:si-epic(sag %fail)
+                      si-cor
```

Comment:
I was thinking about the time limit here: I don't know what is the typical pattern of updates propagating throughout the network. In some cases it might mean hundreds/thousands re-subscriptions to outdated ships every interval. Perhaps for the initial deploy it is worth setting to several hours or even one day? Next time around we can update contacts with a smaller interval.

Reply:
Instead of using `lib-subscriber`, I have implemented my own tiny resubscription logic. `lib-resubscriber` expect subscriptions to be uniquely identified by the wire, which isn't the case for `%contacts`, and would require further changes. We can do that though, if desired.